internal/factory/daemonset.go

```diff
@@ -10,9 +11,13 @@ import (
 //NewDaemonSet stubs an instance of a daemonset
 func NewDaemonSet(daemonsetName, namespace, loggingComponent, component string, podSpec core.PodSpec) *apps.DaemonSet {
 	labels := map[string]string{
-		"provider":      "openshift",
-		"component":     component,
-		"logging-infra": loggingComponent,
+		"provider":                     "openshift",
```

Comment:
The `labels` variable is used for both the labels of the DaemonSet and Pod resources as well as in the `LabelSelector` in the DaemonSet.

Selectors are write-once attributes, so this will create an error when the CLO tries to update an existing DaemonSet, which would make updates from a previous version impossible.

I think you need to split up the labels and the LabelSelector, so that the selector can stay with the current value to avoid this error. Alternatively you need to make sure that the old resource is deleted before the update (I would not go that route, though).